lib/api/Cargo.toml

```diff
@@ -67,6 +67,8 @@ llvm = [
     "wasmer-compiler-llvm",
     "compiler",
 ]
+# enables internal features used by the deprecated API.
+deprecated = []
```

Comment:
Optional: really wish we didn't have to do this. Deprecated ought to be built on top of the API. Or put another way, anything deprecated does should be supported by our real API as something that other library consumer may also want to do.

Not that I'm offering an alternative. IIUC we considered wrapping it in a Mutex and rejected that?

Reply:
> Not that I'm offering an alternative. IIUC we considered wrapping it in a Mutex and rejected that?

That was a different PR, if it were as simple as just conditionally putting something behind a mutex, then I agree.  I tried 2-3 times to make it work without editing the API and this is the only way I've been able to make it work.

I agree that it kind of sucks, but I figure the deprecated API won't be around forever so 🤷